packages/sources/northwestnodes/src/endpoint/staking/ethereum/epoch/single.ts

```diff
@@ -0,0 +1,95 @@
+import { AdapterEndpoint } from '@chainlink/external-adapter-framework/adapter'
+import { InputParameters } from '@chainlink/external-adapter-framework/validation'
+import { HttpTransport } from '@chainlink/external-adapter-framework/transports'
+import { IEpochResponse } from '../../../../responses/IEpochResponse'
+import { makeLogger } from '@chainlink/external-adapter-framework/util'
+import { config } from '../../../../config'
+
+const logger = makeLogger('NWN_EthStakingSingleEpochEndpoint')
+
+const inputParameters = new InputParameters({
+  id: {
+    description: 'The id of the epoch, can be "finalized" or a numeric',
+    type: 'string',
+    required: false,
+    default: 'finalized',
+  },
+})
+
+type EndpointType = {
+  Parameters: typeof inputParameters.definition
+  Settings: typeof config.settings
+  Response: {
+    Data: string
+    Result: null
+  }
+  Provider: {
+    ResponseBody: IEpochResponse
+    RequestBody: null
+  }
+}
+
+const transport = new HttpTransport<EndpointType>({
+  prepareRequests: (params: (typeof inputParameters.validated)[], settings) => {
+    const baseURL = settings.API_ENDPOINT
+    const id = params[0].id.toLowerCase()
```

Comment:
Same thing as the endpoint above, you could have different requests to this adapter that want different epochs so the source requests would be different

Reply:
Fixed